lib/Backend/Commands.hs

```diff
@@ -87,7 +88,27 @@ createCmd (CreateOptions entryPath _edit force tags fields privateFields) = do
       & E.fields .~ HashMap.fromList allFields
       & E.tags .~ tags
 
-  -- TODO: check that a directory does not exist at the given path.
+    checkForEntriesInEntryPath :: Member BackendEffect r => EntryPath -> Sem r (Maybe EntryPath)
+    checkForEntriesInEntryPath entryPath =
+      let 
+        Path.EntryPath (start NE.:| rest) = entryPath
+        startPath = Path.EntryPath (start NE.:| [])
+        restPath = Path.Path rest
```

Comment:
We could simplify this by separating these two bits of logic: 1. build a list of all the paths we need to check and then 2. iterate through them and check if they already point to an entry.

I think we could add a function like this to the `Path` module (which might even be useful in other situations):

```hs
entryPathParentDirs :: EntryPath -> NonEmpty Path
```

such that (in pseudocode):

```hs
entryPathParentDirs "/a/b/c" == [  "/", "/a", "/a/b" ]
```

Then, here in `checkForEntriesInEntryPath`, all we had to do was iterate through those paths.